src/solvers/smt2/smt2_parser.cpp

```diff
@@ -781,11 +781,14 @@ exprt smt2_parsert::function_application()
           else if(id=="sign_extend")
           {
             auto width=to_unsignedbv_type(op[0].type()).get_width();
-            signedbv_typet signed_type(width+index);
+            signedbv_typet small_signed_type(width);
+            signedbv_typet large_signed_type(width + index);
             unsignedbv_typet unsigned_type(width+index);
 
             return typecast_exprt(
-              typecast_exprt(op[0], signed_type), unsigned_type);
+              typecast_exprt(
+                typecast_exprt(op[0], small_signed_type), large_signed_type),
```

Comment:
Not sure why we apply the two casts here (small and large). Could you maybe add a comment?

Reply:
done